docs/network-restrictions.md

```diff
@@ -0,0 +1,45 @@
+# Running behind a proxy
+
+If nodes only have access via a HTTP proxy, you can let the machine-controller configure all new nodes to use this proxy.
+For this the following flag must be set on the machine-controller side:
+```bash
+-node-http-proxy="http://192.168.1.1:3128"
+```
+This will set the following environment variables via /etc/environment on all nodes (lower & uppercase):
+- HTTP_PROXY
+- HTTPS_PROXY
+
+`NO_PROXY` can be configured using a dedicated flag:
+```bash
+-node-no-proxy="10.0.0.0/24"
```

Comment:
Is it possible to do this with CIDR ranges? I always thought no_proxy has to be a suffix which would mean this never matches as no ip address ends with `10.0.0.0/24`

Reply:
Was an assumption as that's supported in the `net` package of golang: https://github.com/golang/net/blob/master/http/httpproxy/proxy.go#L39

cc @thz

Reply:
Okay, but that still doesn't work for the rest of the world, sample:

```
$ curl https://8.8.8.8
curl: (60) SSL certificate problem: self signed certificate
More details here: https://curl.haxx.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
alvaro@t470s-[2019-06-07-12:23]:~
$ export https_proxy=127.0.0.1
alvaro@t470s-[2019-06-07-12:23]:~
$ curl https://8.8.8.8
curl: (7) Failed to connect to 127.0.0.1 port 1080: Connection refused
alvaro@t470s-[2019-06-07-12:23]:~
$ export no_proxy=8.8.0.0/16
alvaro@t470s-[2019-06-07-12:24]:~
$ curl https://8.8.8.8
curl: (7) Failed to connect to 127.0.0.1 port 1080: Connection refused
```

And we use curl to download things and need it to work with a proxy there or not?

Reply:
Good point :+1: 
There is no rfc on how the `NO_PROXY` variable should be used.
Though it seems everyone supports IP's & domain names.

I'll adjust the code & doc to reflect that only IP's & domain names must be specified.